packages/babel-types/src/validators.js

```diff
@@ -233,3 +233,47 @@ export function isImmutable(node: Object): boolean {
 
   return false;
 }
+
+/**
+ * Check if two nodes are equivalent
+ */
+
+export function isNodesEquivalent(a, b) {
+  if (typeof a !== "object" || typeof a !== "object" || a == null || b == null) {
+    return a === b;
+  }
+
+  if (a.type !== b.type) {
+    return false;
+  }
+
+  const fields = Object.keys(t.NODE_FIELDS[a.type] || a.type);
```

Comment:
I removed the outer function since it's in babel-types now and I had to add ` || a.type);` add the end of this because of an issue with checking `File.tokens` since the acorn token types aren't in NODE_FIELDS